### DISCUSSION TITLE
Shape Design Tutorial Error

### DISCUSSION THREAD

**Discussion by peb5234** [su2code/SU2#2308] [Q&A]
I have an issue with the "Constrained shape design of a transonic turbulent airfoil at a cte. CL" tutorial that I have not been able to find a fix for elsewhere. I have been able to successfully run of the design tutorials such as the unconstrained airfoil and unsteady NACA0012 tutorials, but when I try to do the same for this one it throws the following outputs:

Error:
![6_19_24_Error](https://github.com/su2code/SU2/assets/172069980/7276e89d-fe77-4141-8892-4e3db03164f4)

Output:
![6_19_24_Output](https://github.com/su2code/SU2/assets/172069980/51b109ff-a498-42f4-9c4f-931b7998b4f8)

I am running the code using the given command:  
python shape_optimization.py -g DISCRETE_ADJOINT -o SLSQP -f turb_SA_RAE2822.cfg

This is the directory that all of the necessary files are in that were cloned from the SU2 Tutorials GitHub page:
![6_19_24_Directory](https://github.com/su2code/SU2/assets/172069980/5b77f484-734c-46be-b78b-6b416a15fd6f)

I am using the given config files and following the tutorial, and cannot find anything wrong but it will not run properly. I was wondering if there is anything I am doing wrong and if anyone could help me solve this issue. I am running this on a computer cluster, and have ran other tutorials in the same exact manner but this time it seems to not work. Please let me know if I can provide any more helpful information.

**Comment by bigfooted** [comment]
I am not sure what causes such an error, the script runs fine on my machine. 

This error can be caused by the file being opened by python and then being closed by another process before python tries to close it again. Or if you open a file in read mode but write to it instead. Or if you run it in windows, then some windows anti-virus/ransomware protection might interfere with reading and writing of files. 
Maybe start with updating to the latest SU2 version to see if the problem still occurs.
